src/ebay-menu-button/menu-button.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { cloneElement, ComponentProps, FC, useEffect, useRef, useState } from 'react'
+import classnames from 'classnames'
+import { filterByType, findComponent } from '../common/component-utils'
+import { handleEscapeKeydown } from '../common/event-utils'
+import { randomId } from '../common/random-id'
+
+import { EbayMenuButtonItem, EbayMenuButtonLabel, EbayMenuButtonSeparator, EbayMenuButtonVariant } from '.'
+import {
+    EbayButton, EbayButtonProps,
+    EbayIcon, EbayIconButton,
+    EbayMenu, EbayMenuType, EbayMenuChangeEventHandler, EbayMenuSelectEventHandler
+} from '..'
+
+export type EbayMenuButtonProps = {
+    a11yText?: string;
+    className?: string;
+    fixWidth?: boolean;
+    reverse?: boolean;
+    text?: string;
+    type?: EbayMenuType;
+    variant?: EbayMenuButtonVariant;
+    onCollapse?: () => void;
+    onExpand?: () => void;
+    onChange?: EbayMenuChangeEventHandler;
+    onSelect?: EbayMenuSelectEventHandler;
+    expanded?: boolean;
+    noToggleIcon?: boolean;
+    checked?: number;
+    collapseOnSelect?: boolean;
+    // todo: implement the following props
+    prefixId?: string;
+    prefixLabel?: string;
+}
+
+type Props = Omit<EbayButtonProps, 'type' | 'variant'> &
+    Omit<ComponentProps<'button'>, 'type' | 'onChange'> &
+    Omit<ComponentProps<'a'>, 'onChange'> &
+    EbayMenuButtonProps
+
+const EbayMenuButton: FC<Props> = ({
+    type,
+    variant = 'button',
+    className,
+    text = '',
+    fixWidth,
+    reverse,
+    expanded: defaultExpanded,
+    noToggleIcon,
+    checked,
+    collapseOnSelect,
+    a11yText,
+    onClick = () => {},
+    onExpand = () => {},
+    onCollapse = () => {},
+    onChange = () => {},
+    onSelect = () => {},
+    children,
+    ...rest
+}) => {
+    const [expanded, setExpanded] = useState(defaultExpanded)
```

Comment:
expanded -> isExpanded?

Reply:
We usually don't use `is` prefix, like `disabled`, `opened`, cause we always know the type of variables and props due to using typescript. In weak-typed languages adding prefix helps to understand variable's type.